internal/compression/deflate.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compression
+
+import (
+	"compress/flate"
+	"errors"
+	"io"
+
+	"connectrpc.com/connect"
+)
+
+// deflateDecompressor is a thin wrapper around a flate Reader.
+type deflateDecompressor struct {
+	reader io.ReadCloser
+}
+
+func (c *deflateDecompressor) Read(bytes []byte) (int, error) {
+	return c.reader.Read(bytes)
+}
+func (c *deflateDecompressor) Reset(rdr io.Reader) error {
+	r, ok := c.reader.(flate.Resetter)
+	if !ok {
+		return errors.New("deflate reader is not able to be used as a resetter")
```

Comment:
I assume this should never happen in practice, right? Maybe a comment to that effect so that the reader knows this would truly be exceptional and is not expected to be an actual outcome.

Reply:
Yeah it should not. I added a comment there to indicate it's not expected but is just there as a safeguard.